design/pipeline-resources.md

```diff
@@ -0,0 +1,199 @@
+# Resources in YAML
+
+Any external service that is consumed as part of your pipeline is a resource. 
+
+An example of a resource can be another CI/CD pipeline that produces artifacts (say Azure pipelines, Jenkins etc.), code repositories (GitHub, Azure Repos, Git), container image registries (ACR, Docker hub etc.) or package feeds (Azure artifact feed, Artifactor etc.).  
+
+## Why resources?
+
+Resources are defined at one place and can be consumed anywhere in your pipeline. Resources provide you the full traceablity of the services consumed in your pipeline including the branch, version, tags, associated commits and work-items. You can fully automate your DevOps workflow by subscribing to trigger events on your resources.
+
+Resources in YAML represent sources of types pipelines, repositories, containers and packages.
+
+
+### Schema
+
+```yaml
+resources:
+  pipelines: [ pipeline ]  
+  repositories: [ repository ]
+  containers: [ container ]
+  packages: [ package ]
+```
+
+---
+
+## Resources: `pipelines`
+
+If you have a pipeline that produces artifacts, you can consume the artifacts by defining a `pipelines` resource. A pipeline can be another Azure DevOps pipeline or any external pipelines like Jenkins etc.
+
+
+### Schema
+
+```yaml
+resources:        # types: pipelines | repositories | containers | packages
+  pipelines:
+  - pipeline: string  # identifier for the pipeline resource
+    type: enum  # type of the pipeline source like azurePipelines, Jenkins etc. 
```

Comment:
@vtbassmatt is "azurePipelines" the standard keyword/type here? Also, @ashokirla, could we have `azurePipelines` be the default type, so that in the common case you don't have to provide `type` at all?